### DISCUSSION TITLE
Directus fails to load after updating to latest version

### DISCUSSION THREAD

**Discussion by AlexVanK** [directus/directus#16359] [Archive]
Hello: I just updated Directus from v9.18.1 to v9.20.4 and my app stopped working.

I'm seeing this error in the console:
![image](https://user-images.githubusercontent.com/56647386/200670779-f70a73cd-ad0a-4c65-80d6-06f3bf0bf711.png)

Here more details:
![image](https://user-images.githubusercontent.com/56647386/200670835-358140b5-3a24-48cd-9b63-8d559aeb422e.png)

Running Directus on IIS. This is my .env
![image](https://user-images.githubusercontent.com/56647386/200671006-34a5c66d-9c77-41cb-bde1-237aef4c7dd1.png)
![image](https://user-images.githubusercontent.com/56647386/200671106-5d7eff55-f827-4aa4-92b3-4fde7370f4c9.png)

If I add the port information to the .env file, I get a complete different error:
![image](https://user-images.githubusercontent.com/56647386/200671460-3874474a-a5ed-49dc-a353-06d5f75b44c4.png)
![image](https://user-images.githubusercontent.com/56647386/200671498-76876d2e-56d0-42cc-a31e-f6bab6a7a2a0.png)

Note that if I run this locally works fine.

**Comment by connorwinston** [comment]
Did you run the migrations before updating?

**Reply by AlexVanK** [reply]
Wait hang on, did I miss something?
I run these as usual:
![image](https://user-images.githubusercontent.com/56647386/200673370-e10b5515-8c5d-45d1-99e3-b3a00bc63d2e.png)

The DB migration part didn't output anything:
![image](https://user-images.githubusercontent.com/56647386/200673550-30dadfc7-e714-459e-b1bb-4d1623292a0b.png)

**Comment by br41nslug** [comment]
It looks like you're getting 2 different OS permission errors:
- the first `uv_uptime` is giving a permission error when nodejs is trying to fetch uptime
- the second when you changed the port indicates that port 81 is already in use (or the user is blocked from binding to any port by the OS)

These 2 make me think that it may be more windows permission related than directus 😬

**Reply by AlexVanK** [reply]
Yes, that's what I thought... 
But nothing really changed on our Windows machine between the Directus updates, except:
- bumping up Node from v12 to v18...
- bumping up npm from v6 to v8

Also, if I run this locally localhost:8055 via `npx directus start`, everything works fine...

**Reply by paescuj** [reply]
>  * bumping up Node from v12 to v18...

I guess it's related to the Node.js update rather than the one of Directus.

Protip: If possible, do such updates step-by-step so you will immediately see which one causes the issue.

**Comment by AlexVanK** [comment]
I discovered that it has to do with the permissions of a user's Role.
On localhost I can login with any user role and the app works just fine.
Once the app is deployed I can only login with non-administrators...

See this screen capture:

https://user-images.githubusercontent.com/56647386/200943064-778375ba-c377-4a58-9fbd-95a578c73a49.mp4




Any idea what may be causing this?

**Comment by AlexVanK** [comment]
I deployed the application in the hopes that the error wouldn't show up... yet here it is:
![image](https://user-images.githubusercontent.com/56647386/202824680-66f378ac-6b2d-49b8-9ff9-f0ac6a7a49c3.png)

````JSON
{
  "errors": [
    {
      "message": "select `ip_access` from `directus_roles` where `id` = '379fb448-598d-42c0-8405-4e279258f0d7' limit 1 - read ECONNRESET",
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR"
      }
    }
  ]
}
```

**Comment by Csaba19999** [comment]
What was the solution?
